### PR TITLE
add shaders for anaglyph Amber-Blue and Green-Magenta

### PR DESCRIPTION
**Pull request by TijmenW**
https://code.google.com/p/dolphin-emu/issues/detail?id=8388

Please ignore 9470a46a8015fed87f058b03194e33e17b66687a

### CODE REVIEW

**Comment by TijmenW** [comment]
I could only test the G-M because I don't have an Amber-Blue.

**Comment by Tilka** [comment]
Please drop the merge commit; we won't merge it like this.

**Comment by BhaaLseN** [comment]
Are those ```mat3``` supposed to be some sort of aligned? Looks a bit random to me.

**Comment by Sonicadvance1** [comment]
Are we even able to use these constants? They are copywritten under "All rights reserved" which I think we aren't allowed to distribute without explicit say from the original author to do so, which breaks GPL?

**Comment by degasus** [comment]
Mathematic constants aren't copyright-able, so they are fine.

**Comment by degasus** [comment]
Are those shaders tested on both D3D and OGL? iirc they have some other default meaning of row vs vector based matrices :/

**Comment by Tilka** [comment]
You'll have to rebase, not merge.

**Comment by TijmenW** [comment]
I did and I'm sorry for all the garbage. Can I start over?

**Comment by Tilka** [comment]
If you mean open a new PR, there's no need for that. Once you get it right locally, you just need to force-push. If you need help, join #dolphin-dev on freenode.

**Comment by TijmenW** [comment]
@Tilka done :) 
@degasus the shaders are for OpenGL. Does the Direct3d backend have support for Postp.e. shaders?

**Comment by Tilka** [comment]
Can you also fix the alignment?

**Comment by Tilka** [comment]
The way you previously did it was actually the Dolphin Way. It's just that some spaces were missing.

**Comment by Tilka** [comment]
lgtm

**Comment by Tinob** [comment]
@degasus: the different API matrix multiplications differences can be bypassed abstracting the operation. The improvements to the interface I'm working on solves the problem.

**Comment by TijmenW** [comment]
Why is it not merged yet? 
@Tilka approved it more than 24 hours ago. 
(I'm sorry I'm so impatient)

**Comment by Tilka** [comment]
@TijmenW The thing is I don't actually know OpenGL. While we're waiting for someone to properly review this, you could rebase the two commits into one.

File: Data/Sys/Shaders/Anaglyph/dubois-LCD-Amber-Blue.glsl [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+{
+	float4 c0 = SampleLayer(0);
+	float4 c1 = SampleLayer(1);
+	mat3 l = mat3( 1.062,-0.205, 0.299,
+	              -0.026, 0.908, 0.068,
+	              -0.038,-0.173, 0.022);
+	mat3 r = mat3(-0.016,-0.123,-0.017,
+	               0.006, 0.062, 0.017,
+	              -0.094,-0.185, 0.991);
+	SetOutput(float4(c0.rgb * l + c1.rgb * r, c0.a));
```
Comment (Sonicadvance1) [inline Data/Sys/Shaders/Anaglyph/dubois-LCD-Amber-Blue.glsl]: please use 'c0.rgb * 1.0' here or remove the multiplication by 1 entirely.
This implicit conversion will fail on OpenGL ES

Comment (Tilka) [inline Data/Sys/Shaders/Anaglyph/dubois-LCD-Amber-Blue.glsl]: That's the letter L.

Comment (Sonicadvance1) [inline Data/Sys/Shaders/Anaglyph/dubois-LCD-Amber-Blue.glsl]: oh derp

Comment (TijmenW) [inline Data/Sys/Shaders/Anaglyph/dubois-LCD-Amber-Blue.glsl]: Should I change the names to left and right?

**Comment by TijmenW** [comment]
@Tilka O, ok.

To be honest, I copied the dubious shader in 262c3b19ecf9a4c022b3667613f892a5539dd34d, and only edited the matrix constants and the source to avoid messing things up.

File: Data/Sys/Shaders/Anaglyph/dubois-LCD-Amber-Blue.glsl (hunk 1 above)
Comment (Tilka) [inline Data/Sys/Shaders/Anaglyph/dubois-LCD-Amber-Blue.glsl]: Nah, should be okay.

**Pull request merged**